### ISSUE TITLE
system_sdk: macOS 13 support

### ISSUE THREAD

**Issue by kamidev** [hexops/mach#606]
```
➜  mach git:(main) zig build -h
Unsupported Target!
thread 2456260 panic: reached unreachable code
/Users/jonas/src/zig/mach/libs/glfw/system_sdk.zig:204:5: 0x1006ba9b3 in include (build)
    unreachable;
    ^
/Users/jonas/src/zig/mach/libs/glfw/build.zig:157:23: 0x10071a10f in linkGLFWDependencies (build)
    system_sdk.include(b, step, options.system_sdk);
```

```
➜  mach git:(main) sw_vers; zig version
ProductName:		macOS
ProductVersion:		13.0
BuildVersion:		22A380
0.11.0-dev.4737+841235b3f
➜  mach git:(main) git log -1 | tee
commit 8638ec246e89a0647d18631c8fb26a78ebf8b70c
Author: Stephen Gutekanst <stephen@hexops.com>
Date:   Sun Oct 30 10:43:53 2022 -0700

    earcut: switch from pointer linked-list to index linked-list

    * Means we can use a MultiArrayList as the backing array, which solves the lifetime
      issue and enables reuse of the buffer for multiple polygon tesselations.
    * Improves performance.
    * Simplifies accesses, removes optionality except in necessary cases.

    Signed-off-by: Stephen Gutekanst <stephen@hexops.com>
```

**Comment by slimsag** [comment]
Should be fixed now; let me know if you encounter any issues.

**Comment by kamidev** [comment]
I am happy to report that all tests pass using this configuration:

```
➜  mach git:(main) sw_vers; clang --version; zig version
ProductName:		macOS
ProductVersion:		13.0.1
BuildVersion:		22A400
Homebrew clang version 15.0.5
Target: arm64-apple-darwin22.1.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm/bin
0.11.0-dev.5059+30eb2a175
➜  mach git:(main) git log -1 | tee
commit 568d0155fdd388f89a9c29872fdf1223a27445a1
Author: Stephen Gutekanst <stephen@hexops.com>
Date:   Fri Nov 25 10:31:04 2022 -0700

    system-sdk: use macOS 12 SDK for macOS 13 for now

    Fixes hexops/mach#606

    Signed-off-by: Stephen Gutekanst <stephen@hexops.com>
```
